subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/JvmComponentDependencies.java

```diff
@@ -125,4 +125,15 @@ public interface JvmComponentDependencies {
      * @since 7.6
      */
     Dependency testFixtures(ModuleDependency moduleDependency);
+
+    /**
+     * Create a dependency on the current project's complete view. During compile-time, this dependency will
+     * resolve the current project's implementation in addition to its API. During runtime, this dependency
+     * behaves as a usual project dependency.
+     *
+     * @return A new dependency
+     *
+     * @since 7.6
+     */
+    ProjectDependency projectComplete();
```

Comment:
I've been picking on names lately, apologies in advance.

Reading this and seeing the examples, I think this is name misleading, as it implies the usual `project()` dep is somehow incomplete.  I think you'll get newcomers finding it and thinking, "well, if I have the choice, I definitely want to depend on the _complete_ project, right?"  What happened to `projectInternal()` or `projectImplementation()`?

Reply:
+1 for projectImplementation 

other ideas:
- projectForUnitTesting()
- projectWithImplementation()
- clearboxProject()
- unitTestableProject()

Reply:
The naming for this one has been particularly challenging. At this point, we've mostly narrowed down the options to `projectInternalView()` and `projectInternals()`, with the former being the current state of the PR. Below are some of the other options considered along the way and the reasonings for dismissing them:

1. `projectComplete()`: Based off the attribute value used for the `CompileView` which differentiates between `apiElements` and `compileElements`. This could potentially confuse users who would may assume the "complete" view is better than the default one. 
2. `projectImplementation()`: Too easily confused with the `implementation` configuration. Users could assume they are getting dependencies only from the project's `implementation` configuration. This overloads the term "Implementation". 
3. `projectInternal()`: Too easily confused with `ProjectInternal`
4. `projectCompile()`: Can be confused as providing the project's compile classpath and only the compile classpath. This dependency is in reality no different than the `project()` dependency during runtime, so having "compile" in the name seems wrong. 

In response to the additional posted ideas:
- IMO the clearbox/opaquebox naming scheme seems forced. I'd prefer to steer away for now
- I would like to avoid linking this dependency to unit testing, since people will certainly figure out ways to use this dependency for suite types which aren't necessarily unit tests. 

I am still open to ideas, and test suites are still incubating. I am pretty satisfied with `projectInternalView()`, though `projectInternals()` is a close second which I would be content with. Thoughts?

Reply:
I find myself in agreement with the dismissed names, Justin provides good reasons for not using them.

For the two candidates, my preference is on `projectInternalView()`.

One thing that could be done is actually name it `internalProjectView()` so that it is easy to find if you know about it and know you need it. But it also does not come up as a suggestion when you start typing `project`.

I also cannot find an alternative proposal that makes sense at this point.